packages/api-client-core/src/GadgetFunctions.ts

```diff
@@ -206,3 +207,63 @@ export interface GlobalActionFunction<VariablesT> {
 
 export type AnyActionFunction = ActionFunctionMetadata<any, any, any, any, any, any> | GlobalActionFunction<any>;
 export type AnyBulkActionFunction = ActionFunctionMetadata<any, any, any, any, any, true>;
+
+// This is a function that represents a computed view that doesn't take any input parameters/variables.
+// Result is an explicit type parameter defining the shape of the full result.
+export type ComputedViewFunctionWithoutVariables<Result> =
+  // Available, Options and Defaults are inferred at call time.
+  <
+    // Available is the full FieldSelection type derived from the shape of the Result type, i.e. all possible selections.
+    Available extends AvailableSelection2<Result> & FieldSelection,
```

Comment:
Note the `& FieldSelection`, something about `AvailableSelection<>` makes it not assignable to `FieldSelection` so narrowing it down to what matches.